rabbitmq/json-queues.yml

```diff
@@ -0,0 +1,147 @@
+type: json
+data:
+  url: /api/queues
+  server: ${SPM_MONITOR_RABBITMQ_API_URL}
+  basicHttpAuthUsername: ${SPM_MONITOR_RABBITMQ_BASICAUTH_USERNAME}
+  basicHttpAuthPassword: ${SPM_MONITOR_RABBITMQ_BASICAUTH_PASSWORD}
+
+observation:
+  - name: queues
+    metricNamespace: rabbitmq
+    path: $.[?(@.name=${name} && @.vhost=${vhost})]
+
+    metric:
+      - name: queues.consumers.active
+        source: active_consumers
+        type: long_gauge
+        label: active consumers
+        description: Number of active consumers, consumers that can immediately receive any messages sent to the queue
+
+      - name: queues.consumers
+        source: consumers
+        type: long_gauge
+        label: consumers
+        description: Number of consumers
+
+      - name: queues.consumer.utilisation
+        source: consumer_utilisation
+        type: long_gauge
+        label: consumer utilisation
+        description: The ratio of time that a queue's consumers can take new messages
+
+      - name: queues.memory
+        source: memory
+        type: long_gauge
+        label: memory
+        description: Bytes of memory consumed by the Erlang process associated with the queue, including stack, heap and internal structures
+
+      - name: queues.messages
+        source: messages
+        type: long_gauge
+        label: messages
+        description: Count of the total messages in the queue
+
+      - name: queues.messages.rate
+        source: messages_details/rate
+        type: long_gauge
+        label: messages rate
+        unit: message/second
+        description: Count per second of the total messages in the queue
+
+      - name: queues.messages.ready
+        source: messages_ready
+        type: long_gauge
+        label: ready messages
+        description: Number of messages ready to be delivered to clients
+
+      - name: queues.messages.ready.rate
+        source: messages_ready_details/rate
+        type: double_gauge
+        label: ready messages rate
+        unit: message/second
+        description: Number per second of messages ready to be delivered to clients
+
+      - name: queues.messages.unacknowledged
+        source: messages_unacknowledged
+        type: long_gauge
+        label: unacknowledged messages
+        description: Number of messages delivered to clients but not yet acknowledged
+
+      - name: queues.messages.unacknowledged.rate
+        source: messages_unacknowledged_details/rate
+        type: double_gauge
+        label: unacknowledged messages rate
+        unit: message/second
+        description: Number per second of messages delivered to clients but not yet acknowledged
+
+      - name: queues.message_stats.ack
+        source: message_stats/ack
+        type: long_gauge
```

Comment:
Gauge? Same question for some others in this file.

Reply:
counter